src/ts/main/src/main.ts

```diff
@@ -0,0 +1,120 @@
+import {prepend} from "basic/prepend"
+import {PROJECT_PYTHON_FILES} from "./managed-by-build"
+import {CssScope, reduxLoopBasedImprovedLoop, ShadowRootContext} from "app/framework"
+import * as RootModule from "app/root"
+import {InvokeSubscriptionFunction} from "../../app/framework/src/subscriber"
+import {Action, createStore, Store} from "redux"
+import {install, StoreCreator} from "redux-loop"
+import {storageAction} from "app/storage"
+
+declare const languagePluginLoader: Promise<any>
+declare const pyodide: any
+
+function main() {
+    const start = (<any>window).start
+    console.log(new Date().getTime() - start)
+
+    console.log(prepend("hello ", "world, from TypeScript"))
+    console.log(PROJECT_PYTHON_FILES)
+
+    const enhancedCreateStore = createStore as StoreCreator
+    const store: Store<RootModule.RootState, Action> =
+        enhancedCreateStore(
+            RootModule.createReducer({
+                improvedLoop: reduxLoopBasedImprovedLoop,
+            }),
+            RootModule.initRootState(),
+            install())
+
+
+    document.addEventListener("DOMContentLoaded", function () {
+
+        document.body.ondragover = (e) => {
+            e.preventDefault()
+            e.stopPropagation()
+        }
+
+        document.body.ondrop = (e) => {
+            e.preventDefault()
+            e.stopPropagation()
+
+            const fileList = e.dataTransfer!.files
```

Comment:
what does the `!` do?? also - is files a list if someone drags a folder? Should we filter out file types for .py?

Reply:
We should do all of that. This is just kind of a hack for demonstration purposes right now. I'll add a code comment.